client/web-sveltekit/src/routes/search/PreviewPanel.svelte

```diff
@@ -0,0 +1,154 @@
+<script lang="ts" context="module">
+    export interface Range {
+        start: Location
+        end: Location
+    }
+
+    export interface Location {
+        // A zero-based line number
+        line: number
+        // A zero-based column number
+        column: number
+    }
+</script>
+
+<script lang="ts">
+    import { mdiClose } from '@mdi/js'
+    import { from } from 'rxjs'
+
+    import CodeMirrorBlob from '$lib/CodeMirrorBlob.svelte'
+    import { isErrorLike } from '$lib/common'
+    import { getGraphQLClient, mapOrThrow, toGraphQLResult } from '$lib/graphql'
+    import Icon from '$lib/Icon.svelte'
+    import LoadingSpinner from '$lib/LoadingSpinner.svelte'
+    import {
+        getFileMatchUrl,
+        createCodeIntelAPI,
+        type ContentMatch,
+        type PathMatch,
+        type SymbolMatch,
+    } from '$lib/shared'
+    import { settings } from '$lib/stores'
+    import { toReadable } from '$lib/utils/promises'
+    import { Alert } from '$lib/wildcard'
+
+    // TODO: should I be importing this from a page? Seems funky.
+    import {
+        BlobPageQuery,
+        BlobSyntaxHighlightQuery,
+    } from '../[...repo=reporev]/(validrev)/(code)/-/blob/[...path]/page.gql'
+
+    import { getSearchResultsContext } from './searchResultsContext'
+
+    export let result: ContentMatch | SymbolMatch | PathMatch
+
+    const searchResultContext = getSearchResultsContext()
+
+    const client = getGraphQLClient()
+    $: codeIntelAPI = createCodeIntelAPI({
+        settings: setting => (isErrorLike($settings?.final) ? undefined : $settings?.final?.[setting]),
+        requestGraphQL(options) {
+            return from(client.query(options.request, options.variables).then(toGraphQLResult))
+        },
+    })
+
+    const blobStore = toReadable(
+        client
+            .query(BlobPageQuery, {
+                repoName: result.repository,
+                revspec: result.commit ?? '', // TODO: default value?
+                path: result.path,
+            })
+            .then(mapOrThrow(result => result.data?.repository?.commit?.blob ?? null))
+    )
+
+    const highlightStore = toReadable(
```

Comment:
This needs to be reactive

```suggestion
    $: highlightStore = toReadable(
```